### PR TITLE
added routes for notifications

### LINKED ISSUE #99: Add notifications page
screen shot 2017-05-28 at 10 55 44 pm

---
- The above image is just representative. We don't want an exact clone of it since the above UI doesn't look very good.
- Unread located at `/notifications`
- All located at `notifications/all`
- The all/unread buttons should be common to both routes
- The mark all as read should only be visible on the unread page

### PR DESCRIPTION
**Pull request by utkarshnath**
#### Checklist
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Added routes for notifications

Fixes #99

### CODE REVIEW

File: app/templates/components/notification-dropdown.hbs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <div class="ui wide notification popup bottom left transition ">
   <div class="ui basic inverted horizontal segments">
     <div class="ui basic left aligned segment weight-800">
-      <p>{{t 'Notifications'}}</p>
+      <a href="/notifications">{{t 'Notifications'}}</a>
```
Comment (geekyd) [inline app/templates/components/notification-dropdown.hbs]: Please use `link-to` helper here. https://guides.emberjs.com/v2.5.0/templates/links/

**Comment by geekyd** [comment]
@utkarshnath We can use component for each notification item as we are reusing the same code for unread and all notification sections. We should reduce duplicate code with a component.
@niranjan94 What do you think ?

**Comment by niranjan94** [comment]
Instead of a component for each notification, we should use the same template file for both all and index. It's basically the same UI with different dataset. 

@utkarshnath please make the necessary changes.

**Comment by utkarshnath** [comment]
@niranjan94 @geekyd Please review

File: app/routes/notifications/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
```
Comment (CosmicCoder96) [inline app/routes/notifications/index.js]: Please add title via `titleToken()`

File: app/routes/notifications/all.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
```
Comment (CosmicCoder96) [inline app/routes/notifications/all.js]: Please add title via `titleToken()`

File: app/routes/notifications.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
```
Comment (CosmicCoder96) [inline app/routes/notifications.js]: Please make use of `AuthenticatedRouteMixin` to ensure access to logged in users only.And please add title via titleToken()

Comment (utkarshnath) [inline app/routes/notifications.js]: @CosmicCoder96 done

File: app/templates/components/notification-dropdown.hbs [hunk 5]
```diff
@@ -4,7 +4,7 @@
 <div class="ui wide notification popup bottom left transition ">
   <div class="ui basic inverted horizontal segments">
     <div class="ui basic left aligned segment weight-800">
-      <p>{{t 'Notifications'}}</p>
+      {{#link-to 'notifications'}}{{t 'Notifications'}}{{/link-to}}
```
Comment (niranjan94) [inline app/templates/components/notification-dropdown.hbs]: Please put it like this (👇 )  for better readability
```hbs
{{#link-to 'notifications'}}
    {{t 'Notifications'}}
{{/link-to}}
```

File: app/routes/notifications/index.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  titleToken() {
+    return this.i18n.t('Unread');
+  },
+  templateName: 'notifications/all',
```
Comment (niranjan94) [inline app/routes/notifications/index.js]: Cool :)

File: app/templates/components/notification-dropdown.hbs (hunk 5 above)
Comment (utkarshnath) [inline app/templates/components/notification-dropdown.hbs]: @niranjan94 done

File: app/templates/notifications.hbs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<h1 class="ui header">{{t 'Notifications'}}</h1>
+<div class="ui divider"></div>
+<div class="row">
+  <a href="/notifications" class="ui button left attached">{{t 'Unread'}}</a>
```
Comment (geekyd) [inline app/templates/notifications.hbs]: @utkarshnath Great work! Can you please check the unread & all buttons? The text seems to be misaligned.
screenshot from 2017-06-02 09-45-46

Comment (utkarshnath) [inline app/templates/notifications.hbs]: @geekyd Its the property of semantic's attached buttons I think and in my opinion it looks better than 👇
screen shot 2017-06-02 at 2 14 09 pm

Comment (geekyd) [inline app/templates/notifications.hbs]: Yes that looks great!

**Pull request merged**